hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/kafka/TopicAclsFacade.java

```diff
@@ -0,0 +1,242 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2018, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.hops.hopsworks.common.dao.kafka;
+
+import io.hops.hopsworks.common.dao.AbstractFacade;
+import io.hops.hopsworks.common.dao.project.Project;
+import io.hops.hopsworks.common.dao.user.Users;
+
+import javax.ejb.Stateless;
+import javax.persistence.EntityManager;
+import javax.persistence.NoResultException;
+import javax.persistence.PersistenceContext;
+import javax.persistence.Query;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.logging.Logger;
+
+@Stateless
+public class TopicAclsFacade extends AbstractFacade<TopicAcls> {
+  
+  public TopicAclsFacade() {
+    super(TopicAcls.class);
+  }
+  
+  @PersistenceContext(unitName = "kthfsPU")
+  private EntityManager em;
+  
+  private static final Logger LOGGER = Logger.getLogger(TopicAclsFacade.class.getName());
+  
+  @Override
+  protected EntityManager getEntityManager() {
+    return em;
+  }
+  
+  public void removeAclFromTopic(String topicName, Project project) {
+    findByTopicName(topicName)
+      .stream()
+      .filter(acl -> KafkaConst.getProjectNameFromPrincipal(acl.getPrincipal()).equals(project.getName()))
+      .forEach(this::remove);
+  }
+  
+  public List<TopicAcls> findByTopicName(String topicName) {
+    return em.createNamedQuery(
+      "TopicAcls.findByTopicName", TopicAcls.class)
+      .setParameter("topicName", topicName)
+      .getResultList();
```

Comment:
@kai-chi Are we sure this returns an empty list if topic is not found?

Reply:
I ran some tests and it returned an empty list. It works as a regular sql where clause so it's just going to return 0 rows.
Also, before invocation the controller checks if the topic exists.